### PR TITLE
feat: allow usage with mongodb@6.x

### PR DESCRIPTION
**Pull request by addaleax**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -41,7 +41,7 @@
   },
   "peerDependencies": {
     "@mongodb-js/oidc-plugin": "^0.3.0",
-    "mongodb": "^5.4.0",
+    "mongodb": "^5.4.0 || ^6.0.0",
     "mongodb-log-writer": "^1.2.0"
```
Comment (lerouxb) [inline package.json:45]: I did not know that you could use || in there. TIL.

File: src/connect.ts [hunk 2]
```diff
@@ -134,7 +134,7 @@ async function resolveMongodbSrv(uri: string, logger: ConnectLogEmitter): Promis
                 resolutionDetails.push({
                   query: 'SRV', hostname, error: args[0]?.message, wasNativelyLookedUp: wasNativelyLookedUp(args[1])
                 });
-                // eslint-disable-next-line node/no-callback-literal
+                // eslint-disable-next-line n/no-callback-literal
```
Comment (lerouxb) [inline src/connect.ts:137]: Just out of curiosity: why was this renamed?

**Review by lerouxb — APPROVED** [review]

File: src/connect.ts (hunk 2 above)
Comment (addaleax) [reply src/connect.ts:137]: https://github.com/eslint-community/eslint-plugin-n

> forked from [eslint-plugin-node](https://www.npmjs.com/package/eslint-plugin-node) v11.1.0. as the original repository seems [no longer maintained](https://github.com/mysticatea/eslint-plugin-node/issues/300).
